zschema/compounds.py

```diff
@@ -192,8 +192,11 @@ def validate(self, value):
             raise DataValidationException("record is not a dict", str(value))
         for subkey, subvalue in value.items():
             if subkey not in self.definition:
-                raise DataValidationException("%s is not a valid subkey of root",
-                                              subkey)
+                port = Port(subkey)
```

Comment:
It is impossible to run validate on any schema with a "Port" defined without this change, since the key in `definitions` is of type `Port`, and not `string`, but after a `json.loads`, it comes in as a string.